ament_lint_cmake/ament_lint_cmake/cmakelint.py

```diff
@@ -382,6 +386,16 @@ def IsValidFile(filename):
     return filename.endswith('.cmake') or os.path.basename(filename).lower() == 'cmakelists.txt'
 
 def ProcessFile(filename):
+    # Store and then restore the filters to prevent pragmas in the file from persisting.
+    original_filters = list(_lint_state.filters)
+    try:
+        return _ProcessFile(filename)
```

Comment:
Ok. I didn't know that _lint_state might change besides the filter set as that's the only change directly operated upon.